guides/store-retrieve-messages.md

```diff
@@ -0,0 +1,162 @@
+# Retrieve Messages Using Waku Store
+
+When running a web dApp or a mobile phone application,
+internet can be unreliable and disconnect.
+
+[Waku Relay](https://rfc.vac.dev/spec/18/) is a gossip protocol.
+As a user, it means that your peers forward you messages they just received.
+If you cannot be reached by your peers, then messages are not relayed;
+relay peers do **not** save messages for later.
+However, [store](https://rfc.vac.dev/spec/13/) peers do save messages they relay,
+allowing you to retrieve them at any time.
+
+In this guide, we'll review how you can use Waku Store to retrieve messages.
+
+Before starting, you need to choose a _Content Topic_ for your dApp.
+Check out the [how to choose a content topic guide](choose-content-topic.md) to learn more about content topics.
+
+For this guide, we are using a single content topic: `/store-guide/1/news/proto`.
+
+# Installation
+
+You can install [js-waku](https://npmjs.com/package/js-waku) using your favorite package manager:
+
+```shell
+npm install js-waku
+```
+
+# Create Waku Instance
+
+In order to interact with the Waku network, you first need a Waku instance:
+
+```js
+import { Waku } from 'js-waku';
+
+const wakuNode = await Waku.create();
+```
+
+# Connect to Other Peers
+
+The Waku instance needs to connect to other peers to communicate with the network.
+You are free to choose other methods to bootstrap and DappConnect will ship with new bootstrap mechanisms in the future.
+
+For now, the easiest way is to connect to Status' Waku fleet:
+
+```js
+import { getStatusFleetNodes } from 'js-waku';
+const nodes = await getStatusFleetNodes();
+await Promise.all(nodes.map((addr) => waku.dial(addr))); 
+```
+
+# Use Protobuf
+
+We recommend you use [protobuf](https://developers.google.com/protocol-buffers/) for messages.
```

Comment:
Why?

Reply:
Not asking you, I mean the guide should explain

Reply:
First, do you agree with this recommendation?

I took that from @arnetheduck https://forum.vac.dev/t/eth-dm-learnings-and-what-is-next/84/2?u=froy

> Generally, waku is not a very efficient protocol - it trades efficiency to gain privacy and decentralisation - at the very minimum, it would make sense to use a payload that is not as wasteful, as json is - ie either compressed json or preferably protobuf that has widespread usage in protocols already, including that of status itself.

I agree with the reason yet and as there is no standard for compressed JSON, I just went down the protobuf route.

If not, what recommendation would you prefer?
If yes, is the space efficiency the main reason in favor of protobuf?

Reply:
If we don't have strong reasons I don't see the need to be prescriptive :)

I think it is fair to say that you can use what you want, but other example protocols use protobuf for X Y Z reasons, so it might make sense to do the same.

Reply:
Thanks for clarifying